### PR TITLE
Transactions

### PR DESCRIPTION
**Pull request by ethanitosardine**
## Description

In this pr I added:
1. commands
2. errors
3. requests

### CODE REVIEW

File: src/clients/openSea.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  buildTransaction(params: BuyNftParams): Promise<Transaction>;
+}
+
+export class OpenSeaClient implements IOpenSeaClient {
+  logger: winston.Logger;
+  cryptoConfig: CryptoConfig;
+  ethOpenSea: OpenSeaSDK;
+
+  constructor(
+    logger: winston.Logger,
```
Comment (daisy1754) [inline src/clients/openSea.ts:23]: just a thought, for things like logger I find it's more easier to just rely on package import, eg

```
import { logger } from "./logger";
```

where logger package returns proper logger for given request context. That way you don't have to keep passing logger across various classes

passing via constructer is more explicit and probably preferred in some case, so just my opinion here

File: src/clients/openSea.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  async getOrder(nftId: string, assetContractAddress: string) {
+    try {
+      const order = await this.ethOpenSea.api.getOrder({
+        side: 'ask',
+        tokenId: nftId.toString(),
+        assetContractAddress: assetContractAddress
+      });
+      return order;
+    } catch (error) {
+      throw new NftNotFoundError();
```
Comment (daisy1754) [inline src/clients/openSea.ts:41]: NftNotFoundError should probably take error as argument and keep track of stacktrace

see eg https://javascript.info/custom-errors

File: src/types/errors/index.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+export class UnauthorizedError extends Error {
```
Comment (daisy1754) [inline src/types/errors/index.ts:1]: suggestion:

define class like ApplicationError with `constructor(cause: error, name: string, message: string, statusCode: int)`, then define function like `func UnauthorizedError(cause: error): ApplicationError`. That way you don't have to repeat line 7 - 10 multiple times.

**Review by daisy1754 — CHANGES_REQUESTED** [review]

File: src/clients/openSea.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
[the remark this answers is not in the document]
Comment (ethanitosardine) [reply src/clients/openSea.ts:1]: cool cool

**Pull request merged**